Examples/Sources/ChatbotExample/Views/MainChatView.swift

```diff
@@ -0,0 +1,156 @@
+import SwiftCrossUI
+
+// MARK: - Main Chat View
+
+struct MainChatView: View {
+    var viewModel: ChatbotViewModel
+    
+    var body: some View {
+        VStack(spacing: 0) {
+            // Header
+            ChatHeaderView(viewModel: viewModel)
+            
+            // Divider
+            Rectangle()
+                .fill(AppColors.border)
+                .frame(height: 1)
+            
+            // Content area
+            if viewModel.isThreadSelected {
+                ChatContentView(viewModel: viewModel)
+            } else {
+                EmptyThreadView(viewModel: viewModel)
+            }
+            
+            // Error message
+            if let errorMessage = viewModel.errorMessage {
+                ErrorMessageView(message: errorMessage)
+            }
+            
+            // Input area
+            if viewModel.isThreadSelected {
+                ChatInputView(
+                    currentMessage: viewModel.currentMessageBinding,
+                    errorMessage: viewModel.errorMessageBinding,
+                    isLoading: viewModel.isLoading,
+                    messageCount: viewModel.currentThreadMessages.count,
+                    onSend: viewModel.sendMessage
+                )
+            }
+        }
+    }
+}
+
+// MARK: - Chat Header View
+
+struct ChatHeaderView: View {
```

Comment:
You can move the bodies of these small views into the `MainChatView` instead by creating computed properties, similar to `var body`, like so:

```swift
@ViewBuilder
var header: some View {
    HStack { ... }
}
```